subprojects/solver/solver-javasmt/src/main/java/hu/bme/mit/theta/solver/javasmt/JavaSMTExprTransformer.java

```diff
@@ -0,0 +1,1266 @@
+        final FloatingPointType fpSort = FloatingPointType.getFloatingPointType(
+                expr.getFpType().getExponent(),
+                expr.getFpType().getSignificand() - 1);
+//        return floatingPointFormulaManager.fromIeeeBitvector(val, fpSort);
+        return context.getFormulaManager().getUFManager().declareAndCallUF("to_fp",
+                fpSort,
+                val,
+                integerFormulaManager.makeNumber(expr.getType().getExponent()),
+                integerFormulaManager.makeNumber(expr.getType().getSignificand() - 1)
+        );
+    }
+
+    private Formula transformFpToBv(final FpToBvExpr expr) {
+        final FloatingPointFormula op = (FloatingPointFormula) toTerm(expr.getOp());
+
+        return floatingPointFormulaManager.castTo(op, expr.getSgn(), FormulaType.getBitvectorTypeWithSize(expr.getSize()));
+    }
+    /*
+     * Arrays
+     */
+
+    private Formula transformFpToFp(final FpToFpExpr expr) {
+        final FloatingPointFormula op = (FloatingPointFormula) toTerm(expr.getOp());
+
+        return floatingPointFormulaManager.castTo(
+                op,
+                true, // ignored
+                FloatingPointType.getFloatingPointType(expr.getExpBits(), expr.getSignBits() - 1),
+                transformFpRoundingMode(expr.getRoundingMode()));
+    }
+
+    private Formula transformArrayRead(final ArrayReadExpr<?, ?> expr) {
+        final ArrayFormula arrayTerm = (ArrayFormula) toTerm(
+                expr.getArray());
+        final Formula indexTerm = toTerm(expr.getIndex());
+        return arrayFormulaManager.select(arrayTerm, indexTerm);
+    }
+
+    private Formula transformArrayWrite(final ArrayWriteExpr<?, ?> expr) {
+        final ArrayFormula arrayTerm = (ArrayFormula) toTerm(
+                expr.getArray());
+        final Formula indexTerm = toTerm(expr.getIndex());
+        final Formula elemTerm = toTerm(expr.getElem());
+        return arrayFormulaManager.store(arrayTerm, indexTerm, elemTerm);
+    }
+
+    private <T1 extends Formula, T2 extends Formula> Formula transformArrayEq(final ArrayEqExpr<?, ?> expr) {
+        final ArrayFormula<T1, T2> leftOpTerm = (ArrayFormula<T1, T2>) toTerm(expr.getLeftOp());
+        final ArrayFormula<T1, T2> rightOpTerm = (ArrayFormula<T1, T2>) toTerm(expr.getRightOp());
+        return arrayFormulaManager.equivalence(leftOpTerm, rightOpTerm);
+    }
+
+    private Formula transformArrayNeq(final ArrayNeqExpr<?, ?> expr) {
+        return booleanFormulaManager.not(
+                (BooleanFormula) transformArrayEq(ArrayEqExpr.create(expr.getLeftOp(), expr.getRightOp()))
+        );
+    }
+
+    private Formula transformArrayLit(final ArrayLitExpr<?, ?> expr) {
+        throw new JavaSMTSolverException("Array literals not yet supported: " + expr);
+
+    }
+
+    private Formula transformArrayInit(final ArrayInitExpr<?, ?> expr) {
+        throw new JavaSMTSolverException("Array literals not yet supported: " + expr);
+    }
+
```

Comment:
https://github.com/sosy-lab/java-smt/issues/360 will hopefully bring about a fix for these.